onnxruntime/core/optimizer/transpose_optimizer/optimizer_api_impl.cc

```diff
@@ -885,7 +886,7 @@ Status TransformLayoutForEP(Graph& graph, bool& modified, const IExecutionProvid
       // Except for resize and convolution ops, all the other layout sensitive ops only require layout transformation
       // for 0th input and output. For resize, add the other relevant inputs which need conversion. For Conv - layout
       // transformer only converts layout for 0th input, weights should be handled by every EP.
-      if (node->OpType() == "Resize") {
+      if (node->OpType() == "Resize" && node->GetExecutionProviderType() != kXnnpackExecutionProvider) {
```

Comment:
If we use #ifdef like `#if defined(USE_CUDA) || defined(USE_ROCM) || defined(USE_NNAPI)` , NNAPI will be crash.
If we remove  || defined(USE_NNAPI), NNAPI can request Resize for the `scales/sizes` tensor needed transpose.

Reply:
Fixed in PR https://github.com/microsoft/onnxruntime/pull/13412/checks